web/source/osk/oskView.ts

```diff
@@ -1078,6 +1078,20 @@ namespace com.keyman.osk {
      * Description  Wrapper function to add and identify OSK-specific event handlers
      */
     ['addEventListener'](event: string, func: (obj) => boolean) {
```

Comment:
Under the niche circumstance that an event listener may with to detach itself temporarily (to prevent an accidental recursion loop) and reattach itself later, having the buttons pop in and out of existence may be "less than preferred."

There's also the matter of ensuring that _all_ listeners have been removed, rather than just dropping the buttons when only _one_ is removed.  That logic would need to go deeper, into the `keyman.utils` method, which this branch's current implementation is able to bypass.

The scenario is niche enough that the additional complexity may not be worth it.  That's my thought, at least.

Reply:
Yep, agreed, so gave this LGTM. It is certainly a niche scenario, and I think we can wait until someone requests it!